src/test/java/ch/naviqore/utils/cache/EvictionCacheTest.java

```diff
@@ -0,0 +1,139 @@
+package ch.naviqore.utils.cache;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class EvictionCacheTest {
+
+    public static final int CACHE_SIZE = 3;
+    public static final int CONCURRENT_ACCESS_COUNT = 1000;
+
+    @Nested
+    class LRU {
+
+        private EvictionCache<String, String> lruCache;
+
+        @BeforeEach
+        void setUp() {
+            lruCache = new EvictionCache<>(CACHE_SIZE, EvictionCache.Strategy.LRU);
+        }
+
+        @Test
+        void shouldComputeIfAbsent() {
+            assertThat(lruCache.isCached("a")).isFalse();
+            lruCache.computeIfAbsent("a", () -> "apple");
+            assertThat(lruCache.isCached("a")).isTrue();
+            assertThat(lruCache.computeIfAbsent("a", () -> "apple")).isEqualTo("apple");
+        }
+
+        @Test
+        void shouldRemoveLeastRecentlyUsed() {
+            lruCache.computeIfAbsent("a", () -> "apple");
+            lruCache.computeIfAbsent("b", () -> "banana");
+            lruCache.computeIfAbsent("c", () -> "cherry");
+
+            // access "a" again to make it recently used
+            lruCache.computeIfAbsent("a", () -> "apple");
```

Comment:
maybe use () -> "avocado" and check that the cached value is still "apple" (so was not newly generated)

Reply:
Good points, implemented in [6633abc](https://github.com/naviqore/public-transit-service/pull/52/commits/6633abcbbb43b4737067cf923fd0773e04228153)